### PR TITLE
Feature: Swift 3

### PR DESCRIPTION
**Pull request by 3lvis**
## Usage

```ruby
pod 'Networking', git: 'git@github.com:3lvis/Networking.git', branch: 'feature/swift-3'
```

### CODE REVIEW

File: Sources/NSFileManager+Helpers.swift [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+extension FileManager {
+    public func fileExistsAtURL(_ url: URL) -> Bool {
         guard let path = url.path else { fatalError("Couldn't get path for url: \(url)") }
 
-        return fileExistsAtPath(path)
+        return fileExists(atPath: path)
     }
 
-    public func removeFileAtURL(url: NSURL) {
+    public func removeFileAtURL(_ url: URL) {
```
Comment (Elland) [inline Sources/NSFileManager+Helpers.swift]: Hmm, how about `public func removeFile(at url: URL)`?

File: Sources/Networking+DELETE.swift [hunk 2]
```diff
@@ -7,8 +7,9 @@ public extension Networking {
      - parameter completion: A closure that gets called when the DELETE request is completed, it contains a `JSON` object and a `NSError`.
      - returns: The request identifier.
      */
-    public func DELETE(path: String, completion: (JSON: AnyObject?, error: NSError?) -> ()) -> String  {
-        let requestID = self.request(.DELETE, path: path, parameterType: .JSON, parameters: nil, parts: nil, responseType: .JSON) { JSON, headers, error in
+    @discardableResult
```
Comment (Elland) [inline Sources/Networking+DELETE.swift]: Discard all the things! 👑

File: Sources/Networking.swift [hunk 3]
```diff
@@ -133,43 +133,43 @@ public class Networking {
...
      Authenticates using Basic Authentication, it converts username:password to Base64 then sets the Authorization header to "Basic \(Base64(username:password))".
      - parameter username: The username to be used.
      - parameter password: The password to be used.
      */
-    public func authenticate(username username: String, password: String) {
+    public func authenticate(username: String, password: String) {
         let credentialsString = "\(username):\(password)"
-        if let credentialsData = credentialsString.dataUsingEncoding(NSUTF8StringEncoding) {
-            let base64Credentials = credentialsData.base64EncodedStringWithOptions([])
+        if let credentialsData = credentialsString.data(using: String.Encoding.utf8) {
```
Comment (Elland) [inline Sources/Networking.swift]: I think you can do `credentialsString.data(using: .utf8)`, no?

File: Tests/DELETETests.swift [hunk 4]
```diff
@@ -93,11 +93,11 @@ class DELETETests: XCTestCase {
             completed = true
         }
 
-        waitForExpectationsWithTimeout(15.0, handler: nil)
+        waitForExpectations(withTimeout: 15.0, handler: nil)
```
Comment (Elland) [inline Tests/DELETETests.swift]: Should this have `self`?

File: Tests/DELETETests.swift [hunk 5]
```diff
@@ -93,11 +93,11 @@ class DELETETests: XCTestCase {
...
     }
 
     func testCancelDELETEWithID() {
-        let expectation = expectationWithDescription("testCancelDELETE")
```
Comment (Elland) [inline Tests/DELETETests.swift]: Should this have `self`?

**Comment by Elland** [comment]
Finished going over it, whew, that was a long read. 📖 😄 

Looks mostly very good, just a couple of comments. ☝️

File: Sources/NSFileManager+Helpers.swift (hunk 1 above)
Comment (3lvis) [inline Sources/NSFileManager+Helpers.swift]: Good idea

File: Sources/Networking.swift (hunk 3 above)
Comment (3lvis) [inline Sources/Networking.swift]: 👍

File: Tests/DELETETests.swift (hunk 4 above)
Comment (3lvis) [inline Tests/DELETETests.swift]: Sure!

Comment (Elland) [inline Tests/DELETETests.swift]: Self all the things!!once

File: Sources/NSFileManager+Helpers.swift [hunk 6]
```diff
@@ -1,13 +1,13 @@
 import Foundation
 
 extension FileManager {
-    public func fileExistsAtURL(_ url: URL) -> Bool {
+    public func exists(at url: URL) -> Bool {
```
Comment (Elland) [inline Sources/NSFileManager+Helpers.swift]: niiiiice

File: Sources/FormDataPart.swift [hunk 7]
```diff
@@ -36,9 +36,9 @@ public struct FormDataPart {
         body += "Content-Type: \(self.type.contentType)\r\n\r\n"
 
         var bodyData = Data()
-        bodyData.append(body.data(using: String.Encoding.utf8)!)
+        bodyData.append(body.data(using: .utf8)!)
         bodyData.append(self.data)
-        bodyData.append("\r\n".data(using: String.Encoding.utf8)!)
```
Comment (Elland) [inline Sources/FormDataPart.swift]: ✂️

**Comment by Elland** [comment]
💣 nice

**Comment by Elland** [comment]
Let's get this shipped!

**Comment by Elland** [comment]
Wait, should this merge into a separate `swift3` branch? 💭

**Comment by 3lvis** [comment]
Damn, I fucked up. CocoaPods doesn't seem to be working with Xcode 8 (Beta 2)

**Comment by Elland** [comment]
👍

**Comment by aasatt** [comment]
Xcode 8 beta 6 needs significant attention. 😪

**Comment by Elland** [comment]
@aasatt yeah, they keep breaking stuff. Some of it for the worse.

**Comment by aasatt** [comment]
@Elland I agree. This one feels like a step in the wrong direction.

**Comment by Elland** [comment]
@aasatt this is the one that enables the `open` bullshit?

**Comment by 3lvis** [comment]
Should be up to date now. The worst change was the AnyHashable :(

**Pull request merged**